src/config.js

```diff
@@ -764,7 +847,13 @@ class Config {
   // Extended: Get an {Array} of all of the `source` {String}s with which
   // settings have been added via {::set}.
   getSources () {
-    return _.uniq(_.pluck(this.scopedSettingsStore.propertySets, 'source')).sort()
+    const arrs = []
+    arrs.push(_.uniq(_.pluck(this.globalSettings.scopedSettings.propertySets, 'source')))
+    arrs.push(_.uniq(_.pluck(this.projectSettings.scopedSettings.propertySets, 'source')))
+    for (let settingsContext of this.pathSettingsMap.values()) {
+      arrs.push(_.uniq(_.pluck(settingsContext.scopedSettings.propertySets, 'source')))
+    }
+    return (_.uniq(_.flatten(arrs))).sort()
```

Comment:
might be a bit too much, but this can be expressed as a large expression. Personally (with the addition of some comments re: what's going on with pathSettingsMap) this is more readable than building up a bunch of lists over time:

```
return _.unique(_.flatten([
  ...this.globalSettings.scopedSettings.propertySets.map(s => s.source),
  ...this.projectSettings.scopedSettings.propertySets.map(s => s.source),
  [...(_.flatten(this.pathSettingsMap.values().map(context =>
    context.scopedSettings.propertySets.map(s => s.source)
  )))
])).sort()
```

cc @matthewwithanm